Packs/PaloAltoNetworksCortex_v2/Integrations/PaloAltoNetworksCortex_v2/PaloAltoNetworksCortex_v2.py

```diff
@@ -0,0 +1,786 @@
+    :return: a Demisto's entry with all the parsed data
+    """
+    table_name: str = 'traffic'
+    context_transformer_function = traffic_context_transformer
+    table_context_path: str = 'Cortex.Logging.Traffic(val.id === obj.id)'
+    return query_table_logs(table_name, context_transformer_function, table_context_path)
+
+
+def query_threat_logs_command():
+    """
+    The function of the command that queries panw.threat table
+    :return: a Demisto's entry with all the parsed data
+    """
+    query_table_name: str = 'threat'
+    context_transformer_function = threat_context_transformer
+    table_context_path: str = 'Cortex.Logging.Threat(val.id === obj.id)'
+    return query_table_logs(query_table_name, context_transformer_function,
+                            table_context_path)
+
+
+def query_table_logs(table_name: str, context_transformer_function,
+                     table_context_path: str):
+    """
+    This function is a generic function that get's all the data needed for a specific table of Cortex and acts as a
+    regular command function
+    :param table_name: the name of the table in Cortex
+    :param context_transformer_function: the context transformer function to parse the data
+    :param table_context_path: the context path where the parsed data should be located
+    :return: the function return's a Demisto's entry
+    """
+    args = demisto.args()
+    fields, query = build_query(args, table_name)
+    client = Client()
+    results = client.query_loggings(query)
+    outputs = [context_transformer_function(record) for record in results]
+    human_readable = records_to_human_readable_output(fields, table_name, results)
+
+    context_outputs: dict = {table_context_path: outputs}
+    # # TODO handle this
+    # # context_standards_outputs: dict = get_context_standards_outputs(results)
+    # # # merge the two dicts into one dict that outputs to context
+    # # context_outputs.update(context_standards_outputs)
+    #
+    return human_readable, context_outputs, results
+
+
+def build_query(args, table_name):
+    fields = args.get('fields', 'all')
+    fields = '*' if 'all' in fields else fields
+    where = build_where_clause(args)
+    query_start_time, query_end_time = query_timestamp(args)
+    timestamp_limitation = f'(TIME(time_generated) BETWEEN TIME(TIMESTAMP("{query_start_time}")) AND ' \
+                           f'TIME(TIMESTAMP("{query_end_time}"))) '
+    limit = args.get('limit', '5')
+    where += f' AND {timestamp_limitation}' if where else timestamp_limitation
+    query = f'SELECT {fields} FROM `firewall.{table_name}` WHERE {where} LIMIT {limit}'
+    return fields, query
+
+
+def fetch_incidents():
+    last_run = demisto.getLastRun()
+    last_fetched_event_timestamp = last_run.get('lastRun')
+
+    if last_fetched_event_timestamp is not None:
+        last_fetched_event_timestamp = datetime.strptime(last_fetched_event_timestamp, '%Y-%m-%dT%H:%M:%S')
```

Comment:
i prefer we use `dateutil.parser.parse` or `dateparser.parse functions instead` instead of `datetime.strptime` as they handle exceptions better in case the given date_format given does not match